File: examples/python/basic_usage.py
#!/usr/bin/env python3
"""
Basic usage examples for BloxAPI with Python
"""

import requests
from pprint import pprint

# Base URL for your BloxAPI instance
BASE_URL = "http://localhost:5000"

def get_user_info(user_id):
    """Get information about a Roblox user"""
    response = requests.get(f"{BASE_URL}/api/users/{user_id}")
    response.raise_for_status()  # Raise exception for HTTP errors
    return response.json()

def search_users(keyword, limit=10):
    """Search for Roblox users by keyword"""
    params = {
        "keyword": keyword,
        "limit": limit
    }
    response = requests.get(f"{BASE_URL}/api/users/search", params=params)
    response.raise_for_status()
    return response.json()

def get_game_info(universe_id):
    """Get information about a Roblox game by universe ID"""
    response = requests.get(f"{BASE_URL}/api/games/{universe_id}")
    response.raise_for_status()
    return response.json()

def get_game_analytics(universe_id, metric_type, time_frame="past7days"):
    """Get analytics for a Roblox game"""
    params = {
        "time_frame": time_frame
    }
    url = f"{BASE_URL}/api/analytics/games/{universe_id}/{metric_type}"
    response = requests.get(url, params=params)
    response.raise_for_status()
    return response.json()

def search_catalog(keyword, category=None, limit=25):
    """Search the Roblox catalog"""
    params = {
        "keyword": keyword,
        "limit": limit
    }
    if category:
        params["category"] = category
        
    response = requests.get(f"{BASE_URL}/api/catalog/search", params=params)
    response.raise_for_status()
    return response.json()

def get_asset_info(asset_id):
    """Get information about a Roblox asset"""
    response = requests.get(f"{BASE_URL}/api/assets/{asset_id}")
    response.raise_for_status()
    return response.json()

def get_badge_info(badge_id):
    """Get information about a Roblox badge"""
    response = requests.get(f"{BASE_URL}/api/badges/{badge_id}")
    response.raise_for_status()
    return response.json()

def check_content_moderation(content_type, content_id):
    """Check moderation status of Roblox content"""
    url = f"{BASE_URL}/api/moderation/content/{content_type}/{content_id}"
    response = requests.get(url)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    # Example usage

    # Get user information for Roblox user ID 1
    print("Getting user information for Roblox founder...")
    user_info = get_user_info(1)
    pprint(user_info)
    print("\n")

    # Search for users with keyword "Roblox"
    print("Searching for users with keyword 'Roblox'...")
    search_results = search_users("Roblox", limit=5)
    pprint(search_results)
    print("\n")

    # Get game information for Natural Disaster Survival (189707)
    print("Getting game information for Natural Disaster Survival...")
    game_info = get_game_info(189707)
    pprint(game_info)
    print("\n")

    # Get visits analytics for the game
    print("Getting visits analytics for the game...")
    analytics = get_game_analytics(189707, "visits")
    pprint(analytics)
    print("\n")

    # Search for dominus hats in the catalog
    print("Searching for dominus hats in the catalog...")
    catalog_results = search_catalog("dominus", category="hats")
    pprint(catalog_results)
    print("\n")

    # Check moderation status of an asset
    print("Checking moderation status of an asset...")
    moderation_status = check_content_moderation("asset", 1818)
    pprint(moderation_status)

File: examples/javascript/basic_usage.js
/**
 * Basic usage examples for BloxAPI with JavaScript
 */

// Base URL for your BloxAPI instance
const BASE_URL = 'http://localhost:5000';

/**
 * Get information about a Roblox user
 * @param {number} userId - The Roblox user ID
 * @returns {Promise<Object>} User information
 */
async function getUserInfo(userId) {
  const response = await fetch(`${BASE_URL}/api/users/${userId}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Search for Roblox users by keyword
 * @param {string} keyword - The search keyword
 * @param {number} limit - Maximum number of results (default: 10)
 * @returns {Promise<Object>} Search results
 */
async function searchUsers(keyword, limit = 10) {
  const params = new URLSearchParams({
    keyword,
    limit
  });
  const response = await fetch(`${BASE_URL}/api/users/search?${params}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Get information about a Roblox game by universe ID
 * @param {number} universeId - The Roblox universe ID
 * @returns {Promise<Object>} Game information
 */
async function getGameInfo(universeId) {
  const response = await fetch(`${BASE_URL}/api/games/${universeId}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Get analytics for a Roblox game
 * @param {number} universeId - The Roblox universe ID
 * @param {string} metricType - The metric type (e.g., "visits", "revenue")
 * @param {string} timeFrame - Time frame for analytics (default: "past7days")
 * @returns {Promise<Object>} Game analytics
 */
async function getGameAnalytics(universeId, metricType, timeFrame = 'past7days') {
  const params = new URLSearchParams({
    time_frame: timeFrame
  });
  const url = `${BASE_URL}/api/analytics/games/${universeId}/${metricType}?${params}`;
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Search the Roblox catalog
 * @param {string} keyword - The search keyword
 * @param {string|null} category - Optional category filter
 * @param {number} limit - Maximum number of results (default: 25)
 * @returns {Promise<Object>} Search results
 */
async function searchCatalog(keyword, category = null, limit = 25) {
  const params = new URLSearchParams({
    keyword,
    limit
  });
  if (category) {
    params.append('category', category);
  }
  const response = await fetch(`${BASE_URL}/api/catalog/search?${params}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Get information about a Roblox asset
 * @param {number} assetId - The Roblox asset ID
 * @returns {Promise<Object>} Asset information
 */
async function getAssetInfo(assetId) {
  const response = await fetch(`${BASE_URL}/api/assets/${assetId}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Get information about a Roblox badge
 * @param {number} badgeId - The Roblox badge ID
 * @returns {Promise<Object>} Badge information
 */
async function getBadgeInfo(badgeId) {
  const response = await fetch(`${BASE_URL}/api/badges/${badgeId}`);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Check moderation status of Roblox content
 * @param {string} contentType - The content type (e.g., "asset", "game")
 * @param {number} contentId - The content ID
 * @returns {Promise<Object>} Moderation status
 */
async function checkContentModeration(contentType, contentId) {
  const url = `${BASE_URL}/api/moderation/content/${contentType}/${contentId}`;
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  return await response.json();
}

/**
 * Run all examples
 */
async function runExamples() {
  try {
    // Get user information for Roblox user ID 1
    console.log('Getting user information for Roblox founder...');
    const userInfo = await getUserInfo(1);
    console.log(userInfo);
    console.log('\n');
    
    // Search for users with keyword "Roblox"
    console.log('Searching for users with keyword "Roblox"...');
    const searchResults = await searchUsers('Roblox', 5);
    console.log(searchResults);
    console.log('\n');
    
    // Get game information for Natural Disaster Survival (189707)
    console.log('Getting game information for Natural Disaster Survival...');
    const gameInfo = await getGameInfo(189707);
    console.log(gameInfo);
    console.log('\n');
    
    // Get visits analytics for the game
    console.log('Getting visits analytics for the game...');
    const analytics = await getGameAnalytics(189707, 'visits');
    console.log(analytics);
    console.log('\n');
    
    // Search for dominus hats in the catalog
    console.log('Searching for dominus hats in the catalog...');
    const catalogResults = await searchCatalog('dominus', 'hats');
    console.log(catalogResults);
    console.log('\n');
    
    // Check moderation status of an asset
    console.log('Checking moderation status of an asset...');
    const moderationStatus = await checkContentModeration('asset', 1818);
    console.log(moderationStatus);
  } catch (error) {
    console.error('Error running examples:', error);
  }
}

// Run the examples if this file is executed directly (e.g., with Node.js)
if (typeof require !== 'undefined' && require.main === module) {
  runExamples();
}

// Export functions if this file is imported as a module
if (typeof module !== 'undefined') {
  module.exports = {
    getUserInfo,
    searchUsers,
    getGameInfo,
    getGameAnalytics,
    searchCatalog,
    getAssetInfo,
    getBadgeInfo,
    checkContentModeration
  };
}